### PR TITLE
GKE Init script

### LINKED ISSUE #3: Create documentation around GKE Cluster bootstrapping
Instructions on creating a cluster. Could be either in the form of a doc or bootstrap script. We should give two options, a simplified bare minimum way to create a private GKE cluster and a production level setup (more nodes and specific node pools for opensearch)

### PR DESCRIPTION
**Pull request by wajihyassine**
### Description of the change

This PR adds a script to bootstrap a GKE cluster as well as required resources such as the VPC Network, Cloud Router, and Cloud NAT if it does not already exist

### Applicable issues

- fixes #3

### CODE REVIEW

**Comment by wajihyassine** [comment]
@hacktobeer whenever you have a moment if you can review, this is essentially also where you can add the monitoring part. Was in between just having this in a doc because whether we want to manage more scripts for other cloud providers but can always revisit that in the future as this also saves on some other steps when first creating a new project. 

FYI @berggren @aarontp optional reviews -  this will be to deploy the GKE cluster for TS/Turbinia. If we wanted to keep Opensearch separate we can optionally add a section to create a Opensearch node pool (VMs).

File: tools/init-gke.sh [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+# A unique ID used for the OSDFIR Infrastructure cluster name.
+INSTANCE_ID='osdfir-cluster'
+# The region and zone where Turbinia will run.  Note that Turbinia does
+# not currently support multi-zone operation.
+ZONE='us-central1-f'
+REGION='us-central1'
+# VPC network to configure the cluster in. This will be automatically created
+# if it does not already exist.
+VPC_NETWORK='default'
+# Control plane IP range for the control pane VPC. Due to the Turbinia
```
Comment (hacktobeer) [inline tools/init-gke.sh:27]: typo (pane == plane)

File: tools/init-gke.sh [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  read response
+  if [[ $response != "y" && $response != "Y" ]] ; then
+    echo $ERRMSG
+    exit 1
+  fi
+fi
+
+# TODO: Do real check to make sure credentials have adequate roles
+if [[ $( gcloud -q --project $DEVSHELL_PROJECT_ID auth list --filter="status:ACTIVE" --format="value(account)" | wc -l ) -eq 0 ]] ; then
+  echo "No gcloud credentials found.  Use 'gcloud auth login' and 'gcloud auth application-default login' to log in"
```
Comment (hacktobeer) [inline tools/init-gke.sh:84]: we need both?

File: tools/init-gke.sh [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+      gcloud --project $DEVSHELL_PROJECT_ID iam service-accounts create "${SA_NAME}" --display-name "${SA_NAME}"
+      # Grant IAM roles to the service account
+      echo "Grant permissions on service account"
+      gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/compute.instanceAdmin'
+      gcloud projects add-iam-policy-binding $DEVSHELL_PROJECT_ID --member=$SA_MEMBER --role='roles/iam.serviceAccountUser'
+    else
+      echo "The GCP service account $SA_NAME already exists. Skipping creation..."
+    fi
+else
+  echo "--no-turbinia-sa specified. Skipping Turbinia GCP service account creation..."
```
Comment (hacktobeer) [inline tools/init-gke.sh:103]: what is the use for this switch if we check on line 93 it is exists anyway?

File: tools/init-gke.sh [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+# to communicate privately.
+VPC_CONTROL_PANE='172.16.0.0/28' # Set to default
+# The cluster name, number of minimum and maximum nodes, machine type and disk 
+# size of the deployed cluster and nodes within it. If you enable --no-node-autoscale
+# the `CLUSTER_MIN_NODE_SIZE becomes the `MAX_NODE_SIZE`.
+CLUSTER_NAME='osdfir-cluster'
+CLUSTER_MIN_NODE_SIZE='1'
+CLUSTER_MAX_NODE_SIZE='20'
+CLUSTER_MACHINE_TYPE='e2-standard-32'
+CLUSTER_MACHINE_SIZE='200'
```
[the remark this answers is not in the document]
Comment (wajihyassine) [reply tools/init-gke.sh]: Fixed

File: tools/init-gke.sh (hunk 3 above)
Comment (wajihyassine) [reply tools/init-gke.sh:103]: The second one is for if someone was completely trying to disable Turbinia from being deployed (and not create the GCP account) and the first one is the check for if it already exists to not trying to create it.

**Review by hacktobeer — APPROVED** [review]

**Pull request merged**